### PR TITLE
Group devices in get-updates output

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
Oh and sample output on my local system.
```
$ fwupdmgr update
Devices with no available firmware updates:
 • Thunderbolt Controller
 • PC601 NVMe SK hynix 512GB
 • Touchpad
Devices with the latest available firmware version:
 • Package level of Dell dock
 • RTS5413 in Dell dock
 • RTS5487 in Dell dock
 • VMM5331 in Dell dock
 • WD19
 • System Firmware
```

File: src/fu-engine.c [hunk 1]
```diff
@@ -3889,6 +3889,18 @@ fu_engine_sort_devices_by_priority_name (gconstpointer a, gconstpointer b)
...
+	    fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_UPDATABLE)) ||
+	    (!fu_device_has_flag (dev_a, FWUPD_DEVICE_FLAG_SUPPORTED) &&
+	    fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_SUPPORTED)))
+		return -1;
+	if ((fu_device_has_flag (dev_a, FWUPD_DEVICE_FLAG_UPDATABLE) &&
+	    !fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_UPDATABLE)) ||
+	    (fu_device_has_flag (dev_a, FWUPD_DEVICE_FLAG_SUPPORTED) &&
+	    !fu_device_has_flag (dev_b, FWUPD_DEVICE_FLAG_SUPPORTED)))
+		return 1;
+	/* then support by priority, so clients can show parents */
```
Comment (hughsie) [inline src/fu-engine.c:3903]: i'm not super keen on this being in the daemon; at the moment we can say the devices are ordered "by priority then name" but with this change it all gets a bit more fuzzy.

Comment (superm1) [reply src/fu-engine.c]: Adjusted

File: src/fu-util-common.c [hunk 2]
```diff
@@ -1776,3 +1776,24 @@ fu_util_send_report (SoupSession *soup_session,
 	/* success */
 	return TRUE;
 }
+
+gint
+fu_util_sort_devices_by_flags (gconstpointer a, gconstpointer b)
```
Comment (hughsie) [inline src/fu-util-common.c:1781]: I think the convention for this kind of sort function is for a _cb suffix, so `fu_util_sort_devices_by_flags_cb`

**Pull request merged**